src/test/java/com/javatraining/worlddata/dao/JPAWorldDaoTest.java

```diff
@@ -0,0 +1,105 @@
+package com.javatraining.worlddata.dao;
+
+import com.javatraining.worlddata.entity.City;
+import jakarta.persistence.*;
+import org.junit.jupiter.api.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+class JPAWorldDaoTest {
+
+    private static EntityManagerFactory entityManagerFactoryMock;
+    private static WorldDao worldDao;
```

Comment:
worldDao is the only real class in this test, it shoud use a mock of EntityManagerFactory which should be injected to worldDao with @InjectMocks annotation

Reply:
to make this injection possible, EntityManagerFactory in JPAWorldDao cannot be final.

Reply:
also, the test class should be annotated with @ExtendWith(MockitoExtension.class)
(proper dependency is needed)